javalib/src/main/scala/java/time/Duration.scala

```diff
@@ -0,0 +1,234 @@
+package java.time
+
+import scala.collection.JavaConverters._
+
+import java.time.temporal._
+
+final class Duration private(seconds: Long, nanos: Int)
+    extends TemporalAmount with Comparable[Duration]
+    with java.io.Serializable with Preconditions {
+
+  require(nanos >= 0 && nanos <= 999999999, "nanos must be >= 0 and <= 999999999")
```

Comment:
I should say that other methods call the constructor, so depending on the arguments passed to these methods the constructor will throw a `DateTimeException`. In particular, `withNanos(1000000000)` will throw because the parameter is too large as required by the JavaDoc. Are you saying I should move these checks into the relevant methods?